src/components/Dataset/GeneratedUI.js

```diff
@@ -62,84 +62,83 @@ const GeneratedUI = ({settings, ui, changeSettingsCallback}) => {
-                        }
-                        break;
-                    case 'int': {
-                        dataEntryElement = (<Input type='number'/>); /* TODO */
-                        }
-                        break;
-                    case 'select': {
-                        let options = [];
-                        // Build options
-                        data.options.forEach(v => {
-                            // <Select> does not like to display text if the key or value (?) is null.
-                            // So convert null to string "- not selected -".
-                            if (v === null || v === '') {
-                                // console.log("pushing: ", {
-                                //     key: "- not selected -",
-                                //     value: "- not selected -",
-                                //     text: "- not selected -"
-                                // });
-                                options.push({
-                                    key: "- not selected -",
-                                    value: "- not selected -",
-                                    text: "- not selected -"
-                                })
-                            } else {
-                                // console.log("pushing: ", {key: v, value: v, text: v});
+                    // Switch input type based on ui specification
+                    let dataEntryElement;
+                    switch (data.ui_format) {
+                        case 'text': {
+                            dataEntryElement = (<Input
+                                    type='text'
+                                    value={settings[ui_name]}
+                                    onChange={(event, data) => {
+                                        handleStrChange(ui_name, data)
+                                    }} />);
+                            }
+                            break;
+                        case 'textarea': {
+                            dataEntryElement = (<TextArea
+                                    value={settings[ui_name]}
+                                    onChange={(event, data) => {
+                                        handleStrChange(ui_name, data)
+                                    }} />);
+                            }
+                            break;
+                        case 'int': {
+                            dataEntryElement = (<Input type='number'/>); /* TODO */
+                            }
+                            break;
+                        case 'select': {
+                            let options = [];
+                            // Build options
+                            data.options.forEach(v => {
                                 options.push({key: v, value: v, text: v})
+                            });
+                            dataEntryElement = (<Select
+                                // value={settings[ui_name]}
```

Comment:
@mattchrlw value preselects based on the dataset setting value. Might just need to uncomment this. I'll try it later